PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/HashGraph.java

```diff
@@ -0,0 +1,156 @@
+package nl.tudelft.pl2016gr2.model;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+
+public class HashGraph implements SequenceGraph {
+
+  private final HashMap<Integer, GraphNode> nodes;
+  private final HashSet<Integer> rootNodes;
+  private final HashSet<String> genomes;
+
+  /**
+   * Constructs an empty <code>HashGraph</code>.
+   */
+  public HashGraph() {
+    nodes = new HashMap<>();
+    rootNodes = new HashSet<>();
+    genomes = new HashSet<>();
+  }
+
+  /**
+   * Constructs a <code>HashGraph</code> with the specified elements and genomes.
+   *
+   * @param nodes   The nodes of the graph
+   * @param genomes The genomes that are represented in the graph
+   */
+  public HashGraph(Map<Integer, ? extends GraphNode> nodes, Collection<String> genomes) {
+    this.nodes = new HashMap<>(nodes);
+    this.genomes = new HashSet<>(genomes);
+    this.rootNodes = new HashSet<>(parseRootNodes());
```

Comment:
This seems like unnecessary overhead, just let the `parseRootNodes()` method return a `HashSet`.